### PR TITLE
Tieto pedagogisen tuen suunnitelman olemassaolosta opiskelijan tietoihin

### LINKED ISSUE #6664: Opiskelijatiedot ja pedagogisen tuen suunnitelma
Joillekin lukio-opiskelijoille tehdään pedagogisen tuen suunnitelma. Tämän tulisi käydä ilmi jotenkin helposti
- ohjaamossa
- arvioinnissa.
- kurssin käyttäjät

Ehdotuksena on, että opiskelijan nimen perässä olisi esim. joku merkki (ikoni, väripallero tms.), kun hänelle on tehty pedagogisen tuen suunnitelma.

Sen näkisivät vain ne henkilökunnan jäsenet, joilla on oikeus nähdä kyseisen opiskelijan pedagogisen tuen suunnitelma.

Käytännössä se voisi olla joku lätkäkin, jota klikkaamalla pääsisi esim. tarkastelemaan opiskelijalle tehtyä pedagogisen tuen suunnitelmaa.

### PR DESCRIPTION
**Pull request by saaralepisto**
closes #6664

### CODE REVIEW

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/evaluation/EvaluationRESTService.java [hunk 1]
```diff
@@ -1765,5 +1771,13 @@ private RestInterimEvaluationRequest toRestModel(InterimEvaluationRequest interi
         interimEvaluationRequest.getRequestText(),
         interimEvaluationRequest.getArchived());
   }
+  
+  private Boolean getHasPedagogyForm(String studentIdentifier) {
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/evaluation/EvaluationRESTService.java:1775]: This method seems to have been copy/pasted to several classes. Couldn't we simply have...

```
public boolean hasPedagogyForm(String studentIdentifier) {
  return findFormByStudentIdentifier(studentIdentifier) != null;
}

```
...in `PedagogyController` instead?

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/organizationmanagement/rest/OrganizationUserManagementRESTService.java [hunk 2]
```diff
@@ -307,7 +307,8 @@ public Response searchStudents(
             userEntity.getUpdatedByStudent(),
             userEntity.getId(),
             null, // flags
-            organizationRESTModel));
+            organizationRESTModel,
+            null));
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/organizationmanagement/rest/OrganizationUserManagementRESTService.java:311]: I'd prefer `Boolean.FALSE` over `null`, when dealing with boolean values. Same comment applies to multiple classes in this PR.

**Review by pkukkon — CHANGES_REQUESTED** [review]

File: muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/pedagogy/PedagogyController.java [hunk 3]
```diff
@@ -213,5 +213,13 @@ public List<PedagogyFormHistory> listHistory(PedagogyForm form) {
     history.sort(Comparator.comparing(PedagogyFormHistory::getCreated).reversed());
     return history;
   }
+  
+  public Boolean getHasPedagogyForm(String studentIdentifier) {
```
Comment (pkukkon) [inline muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/pedagogy/PedagogyController.java:217]: I'd still like to see this optimized as...

```
public boolean hasPedagogyForm(String studentIdentifier) {
  return findFormByStudentIdentifier(studentIdentifier) != null;
}
```

so Java doesn't have to create variables that are immediately garbage collected after leaving this method.

**Review by pkukkon — APPROVED** [review]
Backend OK.

**Review by avukkonen — APPROVED** [review]

**Pull request merged**